### ISSUE TITLE
Very High Graphics Memory Usage

### ISSUE THREAD

**Issue by AustinJ235** [linuxmint/Cinnamon#3892]
Today as I was looking through my nvidia-settings I noticed that I had extremly high graphics memory usage when I was doing virtually nothing. I closed all my open windows and it was still very high in the 1500mb, it was at 1900 some. Then to test this further I switched to my alternate desktop setup of xfce/openbox and my memory usage graphics wise was back down to 200mb ish. I then proceed by restarting my computer, this still didn't make a difference in the graphics memory issue. I am using a 4k monitor with a 1080p on the side if that may cause issues with this.

![memory](https://cloud.githubusercontent.com/assets/5326321/6102546/fa408308-affd-11e4-89c0-d01b4a5bab1a.png)

Using Cinnamon Version 2.4.6-1

**Comment by lestcape** [comment]
@AustinJ235 please, see your cinnamon process... Is not only a graphics memory issue. The last nightly cinnamon has an improve about memory and is possible that more things need to be done, but is highly important reports the problems with the default cinnamon configuration. Externals extensions are not responsibilities of cinnamon devs.

On my opinion, need to have more priority the system ram, than the Graphics memory and is possible that the solution for one could be the solution to the another.
Regards.

**Comment by AustinJ235** [comment]
I completely removed cinnamon with all configuration. No extensions and no custom themes. All other applications closed. Still around the same graphics memory usage. I think that this much graphics memory usage is not accepable because when I am actually doing things and having like 10-15 windows open the graphics memory is around 1900mb which is incredibly excessive. If I want to run a game there is not any room for that.

In terms of getting the absolute latest version of cinnamon, the aur package is currently broken.

**Comment by lestcape** [comment]
Is extremly dificult detect a problem of memory, on the own code and if this involve a propietary drivers is really more difficult. What could be the cause? Cinnamon really have a more intensive usage of the graphics card than others enviroments, this is a true. You can not compare differents desktops, because they use differents Gl functions and possible different features of the graphics drivers.

As a derivate of Gnome Shell(GS) Cinnamon also could have the same problems of GS and you can found on google several reports of users, related with nvidea propietary graphics. I don't know where could be the problem, but also is possible that could not be responsability of cinnamon (or also GC).

https://devtalk.nvidia.com/default/topic/786024/340-46-4-memory-leak/
https://devtalk.nvidia.com/default/topic/797547/?comment=4430924
https://devtalk.nvidia.com/default/topic/754695/?comment=4245484

Aditional, maybe could help, provide more behaviors for the graphics memory... Like for example, what occurs when:

You restart cinnamon?
You close all windows after a high memory usage, but without restart cinnamon?
You have not windows opened and then you opened only one window?

This maybe could help to localized the problem, but possible could not be find, or not exist really on cinnamon.

**Comment by AustinJ235** [comment]
I have been watching this someone closely the past few hours. Resizing windows is definity making it go up quite a bit. For example if I take gnome-terminal and have it normal size I am resting at about 1875mb. If I fullscren that it goes up to 1897, but that could just be more graphics memory is needed for the space being rendered. Closing a window I can see about a 10mb drop. So it seems that it may be just related to the area being rendered.

I let you know some more observations as I seem them.

**Comment by leigh123linux** [comment]
You claim xfce/openbox uses less, what resolution are you using for these non-hidpi DE's?
Running hidpi resolutions requires nearly double the graphics RAM (nothing is for free!).

**Comment by AustinJ235** [comment]
I found the issue to be with the compositing with xinerama enabled. I found the issue when I was looking through my Xorg.log files. I was giving me a warning say that weird things may happen as a result of having the compositing extension with the xinerama extension enabled. I should note that when I disabled compositing the weird issue of dragging things on my desktop stop. When ever I, for example, dragged a file between applications it would cause the application that I was dragging from to crash with a bad window error.

To further test that it was an issue with compositing causing the high memory usage I when on my Mageia/KDE install and that also had graphics memory also ~900mb.

Now if I disable xinerama and just use randr to set up my monitors the graphics memory is back down to what it should be. However my current monitor configuration requires that I use Xinerama since it allows each xscreen to act as it only "virtual" display more or less in terms of what the desktop is concerned.

My 4k monitor in  order to operate at 60hz has to switch to MST mode which was introduced on Display Port 1.2. MST mode allows daisy chaining monitors together to only utilize one display port connection on the graphics card -- useful when you have limited display port connections. With current standards you are unable to push 4k@60hz since the bandwidth doesn't exist. Monitors at this moment to obtain this act as two daisy chained monitors in MST mode, each being half of  the monitor. In terms of  what the monitor configuration is concerned these act as completely separate monitors. To make them act as one monitor I can use Xinerama.

Xinerama with compositing and I think Xinerama in general is currently unsupported in the nvidia driver at least. So this is the cause for the weird issues that I am having such as the high graphics memory issue. At this moment I am stuck using Xinerama with no compositing as I wait for randr to get an implementation when I can make two monitors act as one since it looks like everybody is switching to randr because we have xrandr and waylandrandr. 

This seems like it may be in the works if you read this mailing list, http://lists.x.org/archives/xorg-devel/2014-January/039987.html

Hope this clarified the state of my issue that I am having. As for it is right now I don't think there is a great solution yet unless there is something I can do with randr to accomplish this.
